### PR TITLE
fix: add background color to navigators

### LINKED ISSUE #1627: Strange screen-flash animation when navigating on Android Expo
Fresh install of the latest client + expo, I'm using Android and Expo Go.

When I navigate from screen to screen just using your demo setup (complete fresh install), I get very strange flashes. Attached a video to show it.

I don't get this when I use the IOS simulator for the same project.

https://user-images.githubusercontent.com/2121331/110413142-78589400-80e1-11eb-9f9e-70b32dbfe036.mp4

```

System
  platform           darwin
  arch               arm64
  cpu                8 cores      Apple M1
  directory          name        ~/Sites/name

JavaScript (and globally-installed packages)
  node               15.9.0       ~/.nvm/versions/node/v15.9.0/bin/node
  npm                7.6.1        ~/.nvm/versions/node/v15.9.0/bin/npm
    npm              7.6.1
    yarn             1.22.10
  yarn               1.22.10      ~/.nvm/versions/node/v15.9.0/bin/yarn

Ignite
  ignite-cli         6.2.0        ~/.npm-packages/bin/ignite
  ignite src         build        ~/.npm-packages/lib/node_modules/ignite-cli/build

Android
  java               1.8.0_282    /usr/bin/java
  android home       -            undefined

iOS
  xcode              12.4
  cocoapods          1.10.1       /usr/local/bin/pod  
```

### PR DESCRIPTION
**Pull request by calendee**
Fixes #1627 white flash.

## Please verify the following:

- [X] `yarn ci:test` **jest** tests pass with new tests, if relevant
- [X] `README.md` has been updated with your changes, if relevant

## Describe your PR
Adds a background color (to match the app) and transparency to the root and main navigators. This prevents the flash of white during navigation animations.

Demo with fix:

https://user-images.githubusercontent.com/57374/116876359-f4eb8880-abe1-11eb-962d-5c609be2d053.mp4

### CODE REVIEW

File: boilerplate/app/navigators/root-navigator.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ const RootStack = () => {
   return (
     <Stack.Navigator
       screenOptions={{
+        cardStyle: { backgroundColor: "#5D2555" },
```
Comment (jamonholmgren) [inline boilerplate/app/navigators/root-navigator.tsx:32]: Could we store this color in the palette?

https://github.com/infinitered/ignite/blob/eadde266f6b46659bac9372056face896d602b61/boilerplate/app/theme/palette.ts

**Review by jamonholmgren — CHANGES_REQUESTED** [review]
This is great! Just one change requested. Thanks so much @calendee.

File: boilerplate/app/navigators/root-navigator.tsx (hunk 1 above)
Comment (calendee) [reply boilerplate/app/navigators/root-navigator.tsx:32]: Yep. I considered doing that, but decided it was outside the scope of this PR as that would require changing it in the main template as well.

However, I'll happily add another commit to this same PR to fix both issues.

Comment (calendee) [reply boilerplate/app/navigators/root-navigator.tsx]: I've just pushed these changes.

I also noticed that the palette file (and other hex colors throughout the app) have inconsistent casing. Would you like me to submit another PR to standardize them? If so, what's your preference - uppercase or lowercase?

Comment (markrickert) [reply boilerplate/app/navigators/root-navigator.tsx]: Thanks! This branch looks good to merge @jamonholmgren. 

I don't think we have a preference on the hex capitalization. people are likely to replace those with their own colors anyways. If you send a PR, we'd probably merge it though :D

**Review by jamonholmgren — APPROVED** [review]

**Comment by jamonholmgren** [comment]
Thank you @calendee !!

**Comment by JayPerfetto** [comment]
Why does this problem keep coming back? I see 3 closed issues for it, and I just opened another after creating the demo app based on the instructions in the readme, having never used Ignite or Expo before.

I must say, adding a "color" to "match" the theme of the demo app does not seem like an appropriate fix - perhaps my knowledge of react-native is a bit naive, but I feel like you shouldn't have to tell the app on every screen what color you chose for the background - this PR doesn't even work properly, its still obvious that a different color (pink) from the background (purple gradient) is being rendered before the actual background of the screen

...seems more like hack execution than a real solution to the problem

**Comment by jamonholmgren** [comment]
@JayPerfetto While I appreciate you pointing out what is a real problem, I'd like to request that you do so with a little more graciousness. This project is free and open source and not a paid commercial product. We have put thousands of hours into it, but some issues (even sometimes obvious ones) will slip through.

You can read more about my open source philosophy in [this Twitter thread](https://twitter.com/jamonholmgren/status/1423357104597790721). Essentially, I'll do my best to try not to block PRs or whatever, but the community helps maintain this boilerplate.

So if you have an idea of how to fix this, I'd be more than happy to review & deploy your PR. Thank you!

**Comment by shellscape** [comment]
https://liberamanifesto.com/

A good read for users of open source software.

**Comment by JayPerfetto** [comment]
I'll do my best to be kinder in reporting any future issues.

I am legitimately concerned that a problem *this* visually apparent has persisted this deep into the product lifecycle of the "#1 React Native boilerplate" solution out there. I appreciate that thousands of hours went into creating this framework, but this issue caused me to think twice about it within the first 5 seconds of using the example. 

As I mentioned in my initial comment, I am not an expert in react-native, and I am legitimately asking if this issue is a product of the Ignite framework, or if I should be evaluating other native app frameworks altogether? 

It should not be deemed offensive to be asking these questions in a highly targeted manor to those with intimate familiarity with the codebase of this project.

**Comment by jamonholmgren** [comment]
I don't take offense at anyone who brings up issues in the open source that I help maintain. But I don't think it's reasonable to notice a visual styling regression and immediately assume that the whole project is deeply flawed in some way. This boilerplate has been used to spin up thousands of React Native apps, from hobby apps to multinational corporations, and in general has received very favorable reviews from those who use it.

As I mentioned, I appreciate the bug report, and perhaps we will find a more suitable way to fix the problem than what we've done in the past. An internal engineer at Infinite Red is taking a look at this in the next week or so.

This particular issue has popped up a few times, it's true. My perspective is that this is a fairly common React Navigation issue ([go search their issues to see](https://github.com/react-navigation/react-navigation/issues?q=is%3Aissue+white+flash)) and we aren't immune from being bitten by it, since we rely on React Navigation ourselves.

I couldn't help but notice that in your Github history you open a lot of issues, but I don't see any recent contributions to the open source you're using. I would like to take this opportunity to encourage you to take ownership of open source and help out by contributing back documentation, code, and even answering others' questions. The more that we all help each other out, the better off we all are. I've been contributing to open source for about a decade now, since before I even used Github, and collaborating with other engineers, learning from them, and maintaining a mutual respect and professionalism is what keeps me coming back.

Have a good day!

**Comment by JayPerfetto** [comment]
You're def not wrong that my code contributions on github are to my employer's private repos, and not to open-source projects. I code full time for a living, and I have a hard time spending what little free time I have outside of my job fixing open source projects. A simple "this is a common issue you can track via React Native Navigation" would have sufficed.

Thanks

**Comment by twerth** [comment]
> You're def not wrong that my code contributions on github are to my employer's private repos, and not to open-source projects. I code full time for a living, and I have a hard time spending what little free time I have outside of my job fixing open source projects. A simple "this is a common issue you can track via React Native Navigation" would have sufficed.
> 
> Thanks

I don't think Jamon understands what you're saying @JayPerfetto. He doesn't seem to realize that you have a job, and that you code at your job, and that job is full time. I get it, I've been there, and it's not easy. When I was doing that I barely could squeeze out a few hours to complain about other people's opensource work they voluntarily gave me for free. I didn't have time to contribute to the community, I had a job! A full-time job too! My boss even expected me to show up... every day! I for one, am totally on your side here Jay.

**Comment by JayPerfetto** [comment]
When you call bug reports 'complaints', you're probably already starting on the wrong foot lol but chill with the flexing, we get it, you're martyrs, thanks for this amazing tool I can't wait to bet the success of my company on it

**Comment by JayPerfetto** [comment]
It's unbelievable that I got this type of reaction from the *C-Suite* at InfiniteRed for simply asking if this super basic recurring cosmetic issue was a core React-Native problem or not.

To lecture me in open source respect and then come back with that is an absolute disgrace from a CEO.

**Comment by twerth** [comment]
> When you call bug reports 'complaints', you're probably already starting on the wrong foot lol but chill with the flexing, we get it, you're martyrs, thanks for this amazing tool I can't wait to bet the success of my company on it

Again, I totally agree with you Jay.  I don't get their problem to be honest. 

For the background problem, here is my fix: I set my app's background to transparent, then let the user choose what they like, by setting the wallpaper to whatever they like. It's the best UX in my opinion, stop forcing colors on users!

I still agree with you that Jamon should fix this, ASAP, and keep it fixed this time. I mean it's not like he has a full-time job coding for a living, like us.

**Comment by twerth** [comment]
> It's unbelievable that I got this type of reaction from the _C-Suite_ at InfiniteRed for simply asking if this super basic recurring cosmetic issue was a core React-Native problem or not.
> 
> To lecture me in open source respect and then come back with that is an absolute disgrace from a CEO.

Come on Jay, I'm hilarious. Look, you didn't just submit a bug report and you know it. You came right out from the gate with snark. We pointed this out nicely, but you couldn't just stop. We're the nicest company around.

**Comment by shellscape** [comment]
lol

**Comment by JayPerfetto** [comment]
Right on - I'll accept fault and I'm genuinely sorry for my rude first comment. Let's start from the beginning lol

Hi! First off, thank you for all the hard work that has inevitably gone into this tool - 

That said, I am still experiencing this issue using the latest CLI version (7.1.1) - I opened a bug but it seems like this PR might be related and I was just wondering, as someone new to react-native, is it normal to have to assign a single hex color to your navigator background? For instance, the example app has a purple-y gradient, but this proposed solution sets the background navigator color to a fixed hex color, which I would expect still would have the same symptom of a "flash" of a slightly different color than the background of the app.

Thanks for your input, legitimately, 
Jason

**Comment by twerth** [comment]
Thank you for that response Jay, seriously.

@harrisrobin knows about this bug, he can explain. I believe it's only in dev, not in a release build

**Comment by harrisrobin** [comment]
@JayPerfetto hey 👋🏽 ,

The issue happens because the background image is not loading fast enough in dev. I just tried running the release build, you'll see that there is no flash (see video below) so you will not be experiencing this in prod. 

If in your app, you are loading a large image then you may experience a flash when running in development mode as development mode is not as performant as the release build. This doesn't really have anything to do with this boilerplate specifically, you would likely experience the same thing in a fresh react-native project. 


https://user-images.githubusercontent.com/3880008/128582864-ffd5c0ed-3c02-4e3c-866c-b00c4d838b81.mov


Hope this helps! 

@jamonholmgren maybe we can replace the image with a gradient at a later point to avoid the flash in dev.

**Comment by JayPerfetto** [comment]
> @JayPerfetto hey 👋🏽 ,
> 
> The issue happens because the background image is not loading fast enough in dev. I just tried running the release build, you'll see that there is no flash (see video below) so you will not be experiencing this in prod.
> 
> If in your app, you are loading a large image then you may experience a flash when running in development mode as the development build is not as performant as the release build. This doesn't really have anything to do with this boilerplate specifically, you would likely experience the same thing in a fresh react-native project.
> 
>  Screen.Recording.2021-08-06.at.8.49.13.PM.mov 
> Hope this helps!
> 
> @jamonholmgren maybe we can replace the image with a linear-gradient or a solid color at a later point to avoid the flash in dev.

You know what - just to show you guys I am not actually *as* big a jerk as this thread makes me look, I'll have a go at that proposed solution for you and run up a PR 😃

**Comment by twerth** [comment]
That's awesome @JayPerfetto , we love the help, and making new friends

**Comment by JayPerfetto** [comment]
https://github.com/infinitered/ignite/pull/1759

Implemented a solution here, but the integration tests aren't liking the addition of my new component - a cursory google of the issue seems to point to babel/jest config changes and I don't feel comfortable making those changes without discussion/review from someone more familiar with the Tests pipeline for this project

Thanks!

**Pull request merged**